documentation/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_bulk_moderation_actions.md

```diff
@@ -0,0 +1,526 @@
+facilitate selecting records for bulk action, we'll update the `search_fields`
+for the main searchbar to search text fields (title, description, tags). We'll
+also add some additional text filters to the sidebar:
+
+- `uuid` - it may still be necessary to search by uuid, so we should preserve
+  this functionality. We include it in a separate filter (rather than keeping it
+  in `search_fields`) for search efficiency.
+- `provider`
+- `creator` - this field should have helptext reminding the user to also filter
+  by provider in order to disambiguate a specific creator.
+
+Additional or more sophisticated filters may be added in the future as
+appropriate.
+
+### DeletedMedia List View
+
+We'll add the `reversed_deindex` action to this list view, allowing maintainers
+to select a list of deindexed records and undo deindexing from this view.
+
+To facilitate easily undoing a particular bulk ModerationDecision, we will add a
+text filter for searching by ModerationDecision id. That allows you to select
+all records that were deleted as part of a particular decision.
+
+### SensitiveMedia List View
+
+We will add the `reversed_mark_sensitive` action to this list view, allowing
+maintainers to select a list of sensitive media and unmark them as sensitive
+from this view.
+
+Similar to the DeletedMedia list view, we will add a filter for
+ModerationDecision id to facilitate easily undoing an entire bulk
+ModerationDecision.
+
+### ModerationDecisions List View
+
+To make it easy to view historical ModerationDecisions, we will add one new list
+view. In the view, decisions will be sorted by creation date, and will display:
+
+- `id`
+- `action`
+- `explanation` - truncated `explanation` column
+- `record_count` - a count for the number of records affected by this decision
+
+This view is useful for getting the id of a moderation decision that you want to
+reverse. Clicking on an individual ModerationDecision will take you to a detail
+view that is entirely read-only, as a moderation decision should not be edited
+after it is created. We will add a custom filter to display only bulk moderation
+decisions, defined as a ModerationDecision with more than one media record.
+
+We will disable the option to add a ModerationDecision object directly.
+
+### Report view
+
+No changes are needed to the Report view. Note that by extending the
+ModerationDecision model rather than adding a new model, when a record is viewed
+from the Report view _all_ related moderation decisions (including bulk actions)
+will automatically be visible to moderators.
+
+## Access control
+
+Because bulk moderation decisions can affect a very large number of records,
+access should be limited to only maintainers. We will use Django's existing
+permissions support to ensure that the relevant actions are only visible to
+maintainers.
+
+If we would like to change this policy in the future to allow moderators to make
+bulk decisions, that can be added easily.
```

Comment:
Should we impose any limit to the max number of works in a single moderation decision? Thinking about the fact that we're more or less adding the ability to "delete all" or at least "delete a very large subset" of works (even if only temporarily until the next reindex).

Is it worth limiting bulk actions to 50 works maximum?

Reply:
You answered this later, never mind!

https://github.com/WordPress/openverse/pull/3719#discussion_r1470452715